crates/vm/backends/mod.rs

```diff
@@ -275,6 +278,65 @@ impl Evm {
             Evm::REVM { state } => revm::extract_all_requests(receipts, state, header),
         }
     }
+
+    pub fn simulate_tx_from_generic(
+        &mut self,
+        tx: &GenericTransaction,
+        header: &BlockHeader,
+        spec_id: SpecId,
+    ) -> Result<ExecutionResult, EvmError> {
+        match self {
+            Evm::REVM { state } => {
+                self::revm::aux::simulate_tx_from_generic(tx, header, state, spec_id)
+            }
+            Evm::LEVM {
+                store_wrapper: _,
+                block_cache: _,
+            } => todo!(),
```

Comment:
I don't agree with leaving a todo here, can we leverage the fact this function returns a `Result<T,E> `and return an Error instead of panicking?

Reply:
I think it'll depend on what behavior we expect for the node here. If we do not want to continue running the node, then we should implement this asap to avoid panicking when running with levm. If we do no want to stop the node and kind of ignore the error, then I agree that this should be handled differently.

Reply:
Fixed in https://github.com/lambdaclass/ethrex/pull/2179/commits/c8093d83e19cf902d4f663527deba050537f5567